### PR TITLE
In the case of upper/lowerBoundBaseFee is updated by governance mechanism

### PR DESCRIPTION
**Pull request by bobpkr**
## Proposed changes

- In the case of governance update, return upper/lower bound base fee instead of parentHeader.BaseFee.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: consensus/misc/kip71_test.go [hunk 1]
```diff
@@ -42,6 +42,35 @@ func TestNextBlockBaseFee(t *testing.T) {
...
+		{750000000000, 30000000000, 25000000000, 30000000000},
+		{750000000000, 20000000000, 25000000000, 25000000000},
+	}
+	for i, test := range tests {
+		config := getTestConfig(common.Big2)
+		config.Governance.KIP71.UpperBoundBaseFee = test.upperBoundBaseFee
+		config.Governance.KIP71.LowerBoundBaseFee = test.lowerBoundBaseFee
+		parent := &types.Header{
+			Number:  common.Big3,
+			GasUsed: config.Governance.KIP71.GasTarget, // parentGasUsed == gasTarget
```
Comment (jimni1222) [inline consensus/misc/kip71_test.go:65]: How about adding the test cases to check the next base fee when `parentGasUsed > gasTarget && parentBaseFee.Cmp(upperBoundBaseFee) == 0` and `parentGasUsed < gasTarget && parentBaseFee.Cmp(lowerBoundBaseFee) == 0`?

Comment (bobpkr) [reply consensus/misc/kip71_test.go]: yes, I missed those cases. thanks.

File: consensus/misc/kip71.go [hunk 2]
```diff
@@ -42,17 +42,24 @@ func NextBlockBaseFee(parentHeader *types.Header, config *params.ChainConfig) *b
...
 	if parentGasUsed > upperGasLimit {
 		parentGasUsed = upperGasLimit
 	}
 	if parentGasUsed == gasTarget {
-		return new(big.Int).Set(parentHeader.BaseFee)
+		return new(big.Int).Set(parentBaseFee)
 	} else if parentGasUsed > gasTarget {
 		// shortcut. If parentBaseFee is already reached upperbound, do not calculate.
-		if parentBaseFee.Cmp(upperBoundBaseFee) >= 0 {
+		if parentBaseFee.Cmp(upperBoundBaseFee) == 0 {
```
Comment (kjeom) [inline consensus/misc/kip71.go:62]: Is it okay? https://github.com/klaytn/klaytn/pull/1477#discussion_r912551052 refers `>=` is right, not `==`

File: consensus/misc/kip71_test.go [hunk 3]
```diff
@@ -42,6 +42,38 @@ func TestNextBlockBaseFee(t *testing.T) {
...
+		nextBaseFee       int64
+	}{
+		{750000000000, 25000000000, 30000000, 750000000000, 750000000000},
+		{700000000000, 25000000000, 30000000, 750000000000, 700000000000},
+		{800000000000, 25000000000, 30000000, 750000000000, 750000000000},
+		{750000000000, 25000000000, 30000000, 25000000000, 25000000000},
+		{750000000000, 30000000000, 30000000, 25000000000, 30000000000},
+		{750000000000, 20000000000, 30000000, 25000000000, 25000000000},
+		{750000000000, 25000000000, 40000000, 750000000000, 750000000000},
+		{750000000000, 25000000000, 20000000, 25000000000, 25000000000},
```
Comment (jimni1222) [inline consensus/misc/kip71_test.go:60]:
```suggestion
		{750000000000, 25000000000, 40000000, 750000000000, 750000000000}, // parentGasUsed(40000000) is bigger than gasTarget(30000000)
		{750000000000, 25000000000, 20000000, 25000000000, 25000000000}, // parentGasUsed(20000000) is bigger than gasTarget(30000000)
```

File: consensus/misc/kip71.go (hunk 2 above)
Comment (jimni1222) [reply consensus/misc/kip71.go:62]: I think this can be changed to `==` after applying L47-L51 changes.

Comment (kjeom) [reply consensus/misc/kip71.go:62]: I got it, Thansk @jimni1222

**Comment by jimni1222** [comment]
@bobpkr Can you check the ci failure?

**Review by jimni1222 — APPROVED** [review]

**Review by kjeom — APPROVED** [review]

**Pull request merged**